### PR TITLE
Fix #12695 : RangeIndex generated with pd.concat with ignore_index as True

### LINKED ISSUE #12695: RangeIndex not generated with pd.concat with ignore_index as True
With version 0.18.0 introducing RangeIndex, should RangeIndex be applied for concat when ignore_index is set to True?

``df = pd.concat([df_a, df_b], ignore_index=True)``

df.index is an instance of Int64Index, not RangeIndex currently in 0.18.0

### PR DESCRIPTION
**Pull request by AbsoluteMSTR**
- [x] closes #12695
- [ ] tests added / passed
- [ ] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry

Replaced generated indexes with RangeIndex, test still needed

### CODE REVIEW

File: pandas/tools/merge.py [hunk 1]
```diff
@@ -1079,8 +1079,7 @@ def _get_concat_axis(self):
             if self.axis == 0:
                 indexes = [x.index for x in self.objs]
             elif self.ignore_index:
-                idx = Index(np.arange(len(self.objs)))
-                idx.is_unique = True  # arange is always unique
+                idx = RangeIndex(len(self.objs))
```
Comment (jreback) [inline pandas/tools/merge.py]: use ``com._default_index(...)``

**Comment by jreback** [comment]
tests! always start with the tests to make sure it fails first!

**Comment by jreback** [comment]
- needs some tests (use the one from the original issue)
- whatsnew entry (bug fix)

**Comment by sinhrks** [comment]
Thanks for the PR:) But I've included the fix in #12846 to refactor existing code. Pls lmk if anything.

**Pull request closed without merging**